### PR TITLE
Add explanation for `enablePlain` in `no-client-credentials` mode when `tokenEndpointUri` is not set

### PR DESCRIPTION
**Pull request by mstruk**
Signed-off-by: Marko Strukelj <marko.strukelj@gmail.com>

### Type of change

_Select the type of your PR_

- Documentation

### Description

This is additional documentation for #4746.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by laidan6000** [comment]
Remember to cherry-pick into Strimzi 0.24.x.

**Review by scholzj — APPROVED** [review]

File: documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc
Comment (laidan6000) [inline documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc:137]:
```suggestion
<8> Set to `true` to enable PLAIN authentication on the listener, which is supported by all clients on all platforms. The Kafka client must enable the PLAIN mechanism and set the `username` and `password`. PLAIN can be used to authenticate either by using the OAuth access token, or the OAuth `clientId` and `secret` (the client credentials). The behavior is additionally controlled by whether `tokenEndpointUri` is specified or not (default is `false`). 
+
If `tokenEndpointUri` is specified and the client sets `password` to start with the string `$accessToken:`, the server interprets the password as the access token and the `username` as the account username. Otherwise, the `username` is interpreted as the `clientId` and the `password` as the client `secret`, which the broker uses to obtain the access token in the client's name.
+
If `tokenEndpointUri` is not specified, the `password` is always interpreted as an access token and the `username` is always interpreted as the account username, which must match the principal id extracted from the token. This is known as 'no-client-credentials' mode because the client must always obtain the access token by itself, and can't use `clientId` and `secret`. 
```

**Review by laidan6000 — CHANGES_REQUESTED** [review]
Looks good. I made a few changes to the call out text.

File: documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc
Comment (laidan6000) [inline documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc:138]: @mstruk - I think number 9 also needs to be changed slightly.

File: documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc
Comment (mstruk) [reply documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc:137]:
```
The behavior is additionally controlled by whether `tokenEndpointUri` is specified or not (default is `false`). 
```

I find the final part, that used to be its own sentence, confusing as it appears to refer to `tokenEndpointUri` when in fact it should refer to `enablePlain`.  I suggest putting it back into its own sentence as is the pattern for all the options.

File: documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc
Comment (mstruk) [reply documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc:138]: @laidan6000 Ah, I see what you mean, it's now optional so 'has to be set' is not ok. 
```suggestion
<9> Additional configuration for PLAIN mechanism to allow clients to authenticate by passing `clientId` and `secret` as `username` and `password` as described in the previous point. If not specified the clients can authenticate over PLAIN only by passing an access token as `password` parameter.
```

Comment (laidan6000) [reply documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc:138]: Looks good.

File: documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc
Comment (laidan6000) [reply documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc:137]: OK, please go ahead and change it.

**Review by laidan6000 — APPROVED** [review]
Looks good.

**Comment by scholzj** [comment]
@strimzi/maintainers This changes the Java code ... so we need another reviewer to review / approve this. Can someone have a look?

File: documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc
Comment (tombentley) [inline documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc:142]: This is sufficiently complicated that I think it would be good to have examples (possibly with pictures) which illustrated these alternatives. I'm not suggesting we should do that in this PR though.

**Review by tombentley — APPROVED** [review]

**Pull request merged**
